main.cpp

```diff
@@ -1,66 +1,76 @@
+/* Copyright (c) 2020 Arm Limited
+*
+* SPDX-License-Identifier: Apache-2.0
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
 #include "mbed.h"
 #include "mbed_error.h"
 #include "mbed_fault_handler.h"
 
-mbed_error_ctx error_ctx;
+#if MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED
+mbed_error_status_t err_status;
 mbed_fault_context_t fault_ctx;
-void generate_bus_fault_unaligned_access();
+static bool reboot_error_happened = false;
 
-static int reboot_error_happened = 0;
-static bool continuous_reboot_test = false;
+void generate_bus_fault_unaligned_access()
+{
+    SCB->CCR |= SCB_CCR_UNALIGN_TRP_Msk;
+    uint32_t inv_addr = 0xAAA3;
+    uint32_t val = *(uint32_t *)inv_addr;
+    printf("val= %X\n", (unsigned int)val);
+    return;
+}
 
-#if MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED
-void mbed_error_reboot_callback(mbed_error_ctx *error_context) {
-    printf("\nmbed_error_reboot_callback: System rebooting, reboot error callback received");
-    reboot_error_happened = 1;
-    mbed_get_reboot_error_info(&error_ctx);
+//   Implementation for Mbed OS fatal error reboot callback
+void mbed_error_reboot_callback(mbed_error_ctx *error_context)
+{
+    reboot_error_happened = true;
```

Comment:
I would rename `reboot_error_happened` to `fault_generated` and set it to true in `generate_bus_fault_unaligned_access`